contracts/common/bases/FeeUpgradeable.sol

```diff
@@ -94,6 +115,31 @@ abstract contract FeeUpgradeable is Initializable, IFee {
         emit PerformanceFeeChange(performanceFee, params);
     }
 
+    function _setTokenPriceHighWatermark(uint256 newTokenPriceHighWaterMark)
+        internal
+        virtual
+    {
+        tokenPriceHighWatermark = newTokenPriceHighWaterMark;
```

Comment:
you can also emit event here

Reply:
Could, but wanted to be consistent with our other fees, where we don't emit events. This might change though, and we might emit events for all.